typing/subst.mli

```diff
@@ -80,10 +89,40 @@ val modtype_declaration:
   scoping -> t -> modtype_declaration -> modtype_declaration
 val module_declaration: scoping -> t -> module_declaration -> module_declaration
 
-(* Composition of substitutions:
-     apply (compose s1 s2) x = apply s2 (apply s1 x) *)
+(** Composition of substitutions:
+     apply (compose s1 s2) x = apply s2 (apply s1 x) **)
 val compose: t -> t -> t
 
+module Unsafe: sig
+
+(** Replacing a module type name S by a non-path signature is unsafe as the
+    packed module type [(module S)] becomes ill-formed. *)
+  val add_modtype: Ident.t -> module_type -> 'any subst -> unsafe
+  val add_modtype_path: Path.t -> module_type -> 'any subst -> unsafe
+
+  (** Deep editing inside a module type require to retypecheck the module, for
+      applicative functors in path and module aliases. *)
+  val add_type_path: Path.t -> Path.t -> 'k subst -> 'k subst
+  val add_type_function:
+    Path.t -> params:type_expr list -> body:type_expr -> 'k subst -> 'k subst
+  val add_module_path: Path.t -> Path.t -> 'k subst -> 'k subst
```

Comment:
I find it confusing that these functions are in Unsafe, but their type remains safety-polymorphic. Did you just forget to return `unsafe`, or is the idea that it would break too much code so you prefer to brand them in the submodule without changing their type for now, or forever?

Reply:
Oops, I forgot to change those types, the interesting point with this move is that it require very little changes.
Concerning, the name I would propose:
```
type unsafe = [`Unsafe]
...
module Unsafe: sig
  type t = unsafe subst
  ...
end
```

Reply:
Yep, this is my favorite proposal as well.